File: LeetCode/LC-1822-Sign-Product-Of -Array/Solution.java
class Solution {
    
    // If we store product of array might exceed memory
    // If any number in array is 0 then product will be 0
    // Better to count negative number if count is even then product will positive

    public int arraySign(int[] nums) {
        int negativeCounter = 0;
        for(int n : nums){
            if (n==0)return 0;
            else if (n < 0){
                negativeCounter++;
            }
        }
        
        return negativeCounter % 2 != 0 ? -1 : 1;
    }
}

File: LeetCode/LC-1822-Sign-Product-Of -Array/Solution.py
class Solution:
    
    def arraySign(self, nums: List[int]) -> int:
        def signFunc(x):
            if x == 0:
                return 0
            elif x < 0 :
                return -1
            else:
                return 1
        mul = 1
        for i in nums:
            mul *= i
        return signFunc(mul)
    